### PR TITLE
Do not show values initially when single select search does not have a filter value

### PR DESCRIPTION
**Pull request by romeovs**
In https://github.com/metabase/metabase/pull/49323 we fixed an issue where the search filter was used in places where a simple dropdown picker would've worked.

But with that we also introduced a small UI inconsistency: when selecting `Search box` now, we always see the full list of values, even when no search terms have been entered.

This PR attempts to fix that by only showing the results when a search value has been entered, but otherwise maintaining the "single value picker" UX.

#### 50 (how things used to work)
https://github.com/user-attachments/assets/dc3d1d82-399a-4805-b1ca-952b899d1413

#### 51 before this PR

https://github.com/user-attachments/assets/213c5400-83e4-4206-996a-420210391757

#### 51 after this PR

https://github.com/user-attachments/assets/1bd6459c-8750-4d9d-9119-227cfa3f9280

### CODE REVIEW

**Comment by mazameli** [comment]
I think this generally makes sense. (Outside the scope of this PR, it makes me wish the search box had type ahead or something: depending on the case, it can be a bit daunting to just have this empty box that says "search" and not really know what values are available.)

File: e2e/test/scenarios/native-filters/native-filters-reproductions.cy.spec.js [hunk 1]
```diff
@@ -1215,3 +1217,49 @@ describe("issue 31606", () => {
...
   });
 });
+
+describe("issue 49577", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  it("not initially show the values when using a single select search box (metabase#49577)", () => {
```
Comment (uladzimirdev) [inline e2e/test/scenarios/native-filters/native-filters-reproductions.cy.spec.js:1227]:
```suggestion
  it("should not show the values initially when using a single select search box (metabase#49577)", () => {
```

**Review by uladzimirdev — COMMENTED** [review]
From code perspective this change LGTM, but from user perspective I'd prefer to see options as well if there are not many of them (e.g. less than half of the available screen).

so I agree with @mazameli here, maybe we can follow google search and show values on click on the search box? 

Screen Recording 2024-11-18 at 11 06 28

**Comment by romeovs** [comment]
@uladzimirdev That's what the dropdown does! IIUC the searchbox is there as an option when users don't want to show any values initially. This is how it used to work with the multi autocomplete input before https://github.com/metabase/metabase/pull/49323 was merged.

@mngr seems to be [on board with these changes](https://metaboat.slack.com/archives/C0645JP1W81/p1731935064069169?thread_ts=1731886412.234039&cid=C0645JP1W81).

**Review by uladzimirdev — APPROVED** [review]

**Review by ranquild — APPROVED** [review]

**Pull request merged**